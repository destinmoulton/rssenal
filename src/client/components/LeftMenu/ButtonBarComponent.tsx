import * as React from "react";

import { Button, Icon, Popup } from "semantic-ui-react";

import LogoutButtonContainer from "../../containers/LeftMenu/LogoutButtonContainer";
import ReorderFoldersModal from "../Modals/ReorderFoldersModal";

export interface IButtonBarComponentProps {
    openAddFeedModal?: (newFeed: object) => void;
    openEditFolderModal?: (folderDefault: object) => void;
    refreshAllFeeds?: () => void;
}

class ButtonBarComponent extends React.Component<IButtonBarComponentProps> {
    constructor(props: IButtonBarComponentProps) {
        super(props);

        this._handleClickRefresh = this._handleClickRefresh.bind(this);
    }

    _handleAddFolder() {
        this.props.openEditFolderModal({ _id: "", name: "" });
    }

    _handleAddFeed() {
        this.props.openAddFeedModal({ _id: "", title: "" });
    }

    _handleClickRefresh() {
        this.props.refreshAllFeeds();
    }

    render() {
        return (
            <div className="rss-leftmenu-buttonbar-container">
                <Button.Group size="mini">
                    <Popup
                        trigger={
                            <Button
                                className="rss-leftmenu-button-addrss"
                                onClick={this._handleAddFeed.bind(this)}
                            >
                                <Icon name="plus" />
                                <Icon name="rss" />
                            </Button>
                        }
                        content="Add RSS Feed"
                    />
                    <Popup
                        trigger={
                            <Button
                                className="rss-leftmenu-button-addfolder"
                                onClick={this._handleAddFolder.bind(this)}
                            >
                                <Icon name="plus" />
                                <Icon name="folder" />
                            </Button>
                        }
                        content="Add Folder"
                    />
                </Button.Group>
                <ReorderFoldersModal />
                <Popup
                    trigger={
                        <Button
                            className="rss-leftmenu-button-reload"
                            icon="refresh"
                            onClick={this._handleClickRefresh}
                            size="mini"
                        />
                    }
                    content="Reload All Feeds"
                />
                <LogoutButtonContainer />
            </div>
        );
    }
}

export default ButtonBarComponent;
