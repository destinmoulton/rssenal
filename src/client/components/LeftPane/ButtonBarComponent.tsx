import debug from "debug";
import * as React from "react";

import { Button, Icon, Popup } from "semantic-ui-react";

import LogoutButtonContainer from "../../containers/LeftPane/LogoutButtonContainer";
import ReorderFoldersModalContainer from "../../containers/Modals/ReorderFoldersModalContainer";

const log = debug("rssenal:ButtonBarComponent");

export interface IButtonBarMapState {}
export interface IButtonBarMapDispatch {
    feedsRefreshAll: () => void;
}

interface IButtonBarProps {
    openAddFeedModal: (newFeed: object) => void;
    openEditFolderModal: (folderDefault: object) => void;
}

type TAll = IButtonBarProps & IButtonBarMapState & IButtonBarMapDispatch;

class ButtonBarComponent extends React.Component<TAll> {
    _handleAddFolder = () => {
        log("_handleAddFolder()");
        this.props.openEditFolderModal({ _id: "", name: "" });
    };

    _handleAddFeed = () => {
        log("_handleAddFeed()");
        this.props.openAddFeedModal({ _id: "", title: "" });
    };

    _handleClickRefresh = () => {
        log("_handleClickRefresh()");
        this.props.feedsRefreshAll();
    };

    render() {
        log("render()");
        return (
            <div className="rss-leftmenu-buttonbar-container">
                <Button.Group size="mini">
                    <Popup
                        trigger={
                            <Button
                                className="rss-leftmenu-button-addrss"
                                onClick={this._handleAddFeed}
                            >
                                <Icon name="plus" />
                                <Icon name="rss" />
                            </Button>
                        }
                        content="Add RSS Feed"
                    />
                    <Popup
                        trigger={
                            <Button
                                className="rss-leftmenu-button-addfolder"
                                onClick={this._handleAddFolder}
                            >
                                <Icon name="plus" />
                                <Icon name="folder" />
                            </Button>
                        }
                        content="Add Folder"
                    />
                </Button.Group>
                <ReorderFoldersModalContainer />
                <Popup
                    trigger={
                        <Button
                            className="rss-leftmenu-button-reload"
                            icon="refresh"
                            onClick={this._handleClickRefresh}
                            size="mini"
                        />
                    }
                    content="Reload All Feeds"
                />
                <LogoutButtonContainer />
            </div>
        );
    }
}

export default ButtonBarComponent;
