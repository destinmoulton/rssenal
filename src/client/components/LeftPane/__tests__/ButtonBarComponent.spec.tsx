import * as React from "react";

import { shallow } from "enzyme";

import ButtonBarComponent from "../ButtonBarComponent";
import { Button } from "semantic-ui-react";
describe("<ButtonBarComponent />", () => {
    it("renders and matches the snapshot", () => {
        const wrapper = shallow(
            <ButtonBarComponent
                feedsRefreshAll={jest.fn()}
                openAddFeedModal={jest.fn()}
                openEditFolderModal={jest.fn()}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
