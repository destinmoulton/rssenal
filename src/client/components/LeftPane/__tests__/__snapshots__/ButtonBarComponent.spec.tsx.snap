// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonBarComponent /> renders and matches the snapshot 1`] = `
<div
  className="rss-leftmenu-buttonbar-container"
>
  <ButtonGroup
    size="mini"
  >
    <Popup
      content="Add RSS Feed"
      keepInViewPort={true}
      on="hover"
      position="top left"
      trigger={
        <Button
          as="button"
          className="rss-leftmenu-button-addrss"
          onClick={[Function]}
          role="button"
        >
          <Icon
            as="i"
            name="plus"
          />
          <Icon
            as="i"
            name="rss"
          />
        </Button>
      }
    />
    <Popup
      content="Add Folder"
      keepInViewPort={true}
      on="hover"
      position="top left"
      trigger={
        <Button
          as="button"
          className="rss-leftmenu-button-addfolder"
          onClick={[Function]}
          role="button"
        >
          <Icon
            as="i"
            name="plus"
          />
          <Icon
            as="i"
            name="folder"
          />
        </Button>
      }
    />
  </ButtonGroup>
  <Connect(ReorderFoldersModalComponent) />
  <Popup
    content="Reload All Feeds"
    keepInViewPort={true}
    on="hover"
    position="top left"
    trigger={
      <Button
        as="button"
        className="rss-leftmenu-button-reload"
        icon="refresh"
        onClick={[Function]}
        role="button"
        size="mini"
      />
    }
  />
  <Connect(LogoutButtonComponent) />
</div>
`;
